snippets/fuzzySearch.md

```diff
@@ -0,0 +1,28 @@
+### fuzzySearch
+
+Determines if the `patrn` matches with `str`
+
+Loops through `str` and determines if it contains all characters of `patrn` and in the correct order. Both the strings are converted to lower case.
+
+Taken from [here](https://github.com/forrestthewoods/lib_fts/blob/80f3f8c52db53428247e741b9efe2cde9667050c/code/fts_fuzzy_match.js#L18).
+``` js
+fuzzySearch = (pattern, string) => { 
+    let patternLength = pattern.length;
+    let strLength = string.length;
+
+    for (var patternIdx = 0, strIdx = 0;patternIdx !== patternLength && strIdx !== strLength;strIdx++){
```

Comment:
I'm pretty certain this whole loop could just be a single `Array.reduce()` with some creative trickery. I'm gonna try my hand at it and let you know if it's doable and readable.